.github/workflows/ci.yml

```
@@ -0,0 +1,131 @@
+name: Continuous Integration (CI)
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+
+jobs:
+  # https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs
+  # Workflows that would otherwise be triggered using `on: push` or
+  # `on: pull_request` won't be triggered if you add any of the
+  # following strings to the commit message in a push, or the HEAD
+  # commit of a pull request:
+  # - [skip ci]
+  # - [ci skip]
+  # - [no ci]
+  # - [skip actions]
+  # - [actions skip]
+
+  test:
+    runs-on: ubuntu-latest
+
+    env:
+      MIX_ENV: test
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      LATEST_TYPESENSE: '27.1'
+
+    strategy:
+      matrix:
+        include:
+        - typesense: '0.25.2'
+          otp: '25'
+          elixir: '1.14'
+          lint: true
+        - typesense: '26.0'
+          otp: '25'
+          elixir: '1.14'
+          lint: true
+        - typesense: '27.1'
+          otp: '25'
+          elixir: '1.14'
+          lint: true
+        - typesense: '0.25.2'
+          otp: '27'
+          elixir: '1.17'
+          lint: true
+        - typesense: '26.0'
+          otp: '27'
+          elixir: '1.17'
+          lint: true
+        - typesense: '27.1'
+          otp: '27'
+          elixir: '1.17'
+          lint: true
```

```suggestion
        - typesense: '0.25.2'
          otp: '25'
          elixir: '1.14'
          lint: false
        - typesense: '26.0'
          otp: '25'
          elixir: '1.14'
          lint: false
        - typesense: '27.1'
          otp: '25'
          elixir: '1.14'
          lint: false
        - typesense: '0.25.2'
          otp: '27'
          elixir: '1.17'
          lint: false
        - typesense: '26.0'
          otp: '27'
          elixir: '1.17'
          lint: false
        - typesense: '27.1'
          otp: '27'
          elixir: '1.17'
          lint: true
```

@jaeyson You shouldn't set lint to all matrix combination, only for `ONE` combination of latest greatest Elixir/Erlang OTP version. Perhaps this explains the `mix format` error you're facing?

---

@kianmeng wow that helped! Thanks alot! 🙏🏾🙏🏾